src/python/pants/backend/python/tasks2/setup_py.py

```diff
@@ -51,6 +51,19 @@ def __init__(self, source_dir, setup_command, **kw):
     self.__setup_command = setup_command.split()
     super(SetupPyRunner, self).__init__(source_dir, **kw)
 
+  def mixins(self):
+    mixins = super(SetupPyRunner, self).mixins().copy()
+    for (key, version) in self._interpreter.extras:
+      if key == 'setuptools':
```

Comment:
Perhaps this should just loop over all extras and require them as mixins, asserting setuptools since this installer is - in particular - setup.py focused. If so, a follow-up change could fold this up into InstallerBase in pex, which has a handle on the interpreter.